loki/backend/cgen.py

```diff
@@ -183,7 +194,11 @@ def visit_Subroutine(self, o, **kwargs):
 
         # Fill the body
         body += [self.visit(o.body, **kwargs)]
-        body += [self.format_line('return 0;')]
+        # body += [self.format_line('return 0;')]
+
+        # if something to be returned, add 'return <var>' statement
+        if return_var is not None:
+            body += [self.format_line(f'return {return_var.name.lower()};')]
```

Comment:
With the above change, this would then become:
```suggestion
        if o.result_name is not None:
            body += [self.format_line(f'return {o.result_name.lower()};')]
```